### PR TITLE
[code-infra] Fix proptypes generation

### LINKED ISSUE #11679: [charts] Pie chart tooltip runtime warning
### Steps to reproduce

Hi, 
i'm using MUI-Charts pie chart.
on runtime, when the tooltip is active (hovering above the chart) i get a **warning**:

`Warning: Failed prop type: Invalid prop 'series.data[0]' of type 'object' supplied to 'DefaultChartsItemTooltipContent', expected 'number'.`

here is part of my code:
` <PieChart
                series={[
                    {
                        data:  testData,
                        valueFormatter(item) {
                            return '$' + item.value.toFixed(2);
                        },
                    },
                ]}
                slotProps={{ legend: { hidden: true } }}
                width={400}
                height={200}
            />`

the testData is `{id: number, label: string, value: number, color: string}`, no nulls.

by trying to fix the problem i changed the testData to be an array of number, but then i get a syntax error.
i read the documentation and didn't find anything to fix it. 

THE CODE IS WORKING  but i can't get rid of the warning.
any ideas?

### Current behavior

_No response_

### Expected behavior

_No response_

### Context

_No response_

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: tooltip runtime warning

### PR DESCRIPTION
**Pull request by alexfauquette**
Verify proptypes update in https://github.com/mui/material-ui/pull/40617

Fix #11325 
Fix #11679

I only spot a differences in the gird. I copy/past those diff in a dedicated file to clean the differences and check what are the different parameters. Here is the result: 

- image one compare the first shape with the second one
- image two compare the first shape with the third one

Screenshot from 2024-01-22 14-06-35
Screenshot from 2024-01-22 14-05-51

### CODE REVIEW

File: packages/grid/x-data-grid/src/components/GridRow.tsx [hunk 1]
```diff
@@ -588,128 +588,394 @@ GridRow.propTypes = {
   onMouseLeave: PropTypes.func,
   pinnedColumns: PropTypes.shape({
```
Comment (flaviendelangle) [inline packages/grid/x-data-grid/src/components/GridRow.tsx:589]: We could probably add this prop to the one to skip :see_no_evil:

File: packages/x-charts/src/ChartsTooltip/DefaultChartsAxisTooltipContent.tsx [hunk 2]
```diff
@@ -72,37 +72,290 @@ DefaultChartsAxisTooltipContent.propTypes = {
...
-    tickInterval: PropTypes.oneOfType([PropTypes.oneOf(['auto']), PropTypes.array, PropTypes.func]),
-    tickLabelInterval: PropTypes.oneOfType([PropTypes.oneOf(['auto']), PropTypes.func]),
-    tickLabelStyle: PropTypes.object,
-    tickMaxStep: PropTypes.number,
-    tickMinStep: PropTypes.number,
-    tickNumber: PropTypes.number.isRequired,
-    tickSize: PropTypes.number,
-    valueFormatter: PropTypes.func,
-  }).isRequired,
+  axis: PropTypes.oneOfType([
```
Comment (flaviendelangle) [inline packages/x-charts/src/ChartsTooltip/DefaultChartsAxisTooltipContent.tsx:75]: So, does it make sense to resolve this prop?

File: packages/x-charts/src/ChartsTooltip/DefaultChartsAxisTooltipContent.tsx [hunk 3]
```diff
@@ -132,45 +385,154 @@ DefaultChartsAxisTooltipContent.propTypes = {
    * The series linked to the triggered axis.
    */
   series: PropTypes.arrayOf(
```
Comment (flaviendelangle) [inline packages/x-charts/src/ChartsTooltip/DefaultChartsAxisTooltipContent.tsx:387]: Same

File: docs/scripts/generateProptypes.ts [hunk 4]
```diff
@@ -43,6 +43,12 @@ async function generateProptypes(project: XTypeScriptProject, sourceFile: string
       if (propsToNotResolve.includes(name)) {
         return false;
       }
+      if (project.name.includes('x-charts') && sourceFile.includes('AxisTooltip')) {
```
Comment (flaviendelangle) [inline docs/scripts/generateProptypes.ts:46]: Could we also include `ChartsContainer` here?
Is there a reason not to skip those props on all components or at least all `x-charts` components?

File: packages/grid/x-data-grid/src/components/GridRow.tsx (hunk 1 above)
Comment (cherniavskii) [reply packages/grid/x-data-grid/src/components/GridRow.tsx:589]: Yes, absolutely 👍🏻

File: docs/scripts/generateProptypes.ts (hunk 4 above)
Comment (flaviendelangle) [reply docs/scripts/generateProptypes.ts:46]: But this prop-type is so long :sob:

Comment (alexfauquette) [reply docs/scripts/generateProptypes.ts:46]: What's the issue of too long proptypes?

Comment (flaviendelangle) [reply docs/scripts/generateProptypes.ts:46]: Bundle size in dev mostly (I hope everybody is removing them from the prod bundle).

Comment (alexfauquette) [reply docs/scripts/generateProptypes.ts:46]: I can reassure you, MUI is automatically removing proptypes in the build thanks to a [babel plugin](https://github.com/oliviertassinari/babel-plugin-transform-react-remove-prop-types) made by a nice contributor 😁

Here is the result if you look in the built package

```js
process.env.NODE_ENV !== "production" ? BarElement.propTypes = {
  // ----------------------------- Warning --------------------------------
  // | These PropTypes are generated from the TypeScript type definitions |
  // | To update them edit the TypeScript types and run "yarn proptypes"  |
  // ----------------------------------------------------------------------
  classes: _propTypes.default.object,
  dataIndex: _propTypes.default.number.isRequired,
  highlightScope: _propTypes.default.shape({
    faded: _propTypes.default.oneOf(['global', 'none', 'series']),
    highlighted: _propTypes.default.oneOf(['item', 'none', 'series'])
  }),
  /**
   * The props used for each component slot.
   * @default {}
   */
  slotProps: _propTypes.default.object,
  /**
   * Overridable component slots.
   * @default {}
   */
  slots: _propTypes.default.object
} : void 0;
```

Comment (alexfauquette) [reply docs/scripts/generateProptypes.ts]: I moved with the remove all big proptypes. Will put them back if it solves some issues asked by users

Comment (flaviendelangle) [reply docs/scripts/generateProptypes.ts]: OK this plugin is awesome :star_struck:

**Comment by alexfauquette** [comment]
Integrated in https://github.com/mui/mui-x/pull/11800

**Pull request closed without merging**